pandas/_libs/groupby.pyx

```diff
@@ -16,14 +16,18 @@ from numpy cimport (ndarray,
 from libc.stdlib cimport malloc, free
 
 from util cimport numeric, get_nat
-from algos cimport swap
-from algos import take_2d_axis1_float64_float64, groupsort_indexer
+from algos cimport (swap, TiebreakEnumType, TIEBREAK_AVERAGE, TIEBREAK_MIN,
+                    TIEBREAK_MAX, TIEBREAK_FIRST, TIEBREAK_DENSE)
+from algos import take_2d_axis1_float64_float64, groupsort_indexer, tiebreakers
 
 cdef int64_t iNaT = get_nat()
 
 cdef double NaN = <double> np.NaN
 cdef double nan = NaN
 
+cdef extern from "numpy/npy_math.h" nogil:
```

Comment:
why do you need these? you can just use ``np.isnan`` no?

Reply:
I was trying to do the nan check inside the nogil block hence the need for it, but your comment makes me realize that I already am creating a mask array that I can reference instead of doing that check again within the nogil.